server/src/analyser.ts

```diff
@@ -689,4 +1042,18 @@ export default class Analyzer {
 
     return tree.rootNode.descendantForPosition({ row: line, column })
   }
+
+  private nodeAtPoints(
```

Comment:
Ah, when it comes to `nodeAtPoints`, the main difference between it and `nodeAtPoint` is that it can return a node for a whole range. For example, given:

```bash
func() {
  echo "Hello"
}
```

Using `nodeAtPoint` at the start of `func` would return the `word` `func`. On the other hand, using `nodeAtPoints` giving it the start and end of `func` would give the whole `function_definition`.

One place I use this is in `findOccurrencesWithin` to get the scope node given a `Range`.

Is this functionality already available somewhere else and I ended up reimplementing it? Or is there a better way of getting the whole node given a range?

Reply:
I now see the type signature is different. All good then. :)